homeassistant/helpers/template.py

```diff
@@ -753,6 +758,98 @@ def __repr__(self) -> str:
         return f"<template TemplateState({self._state.__repr__()})>"
 
 
+class Areas:
+    """Class to expose areas."""
+
+    def __init__(self, hass):
+        """Initialize the area list."""
+        self._hass = hass
+        self._area_registry = hass.data.get(area_registry.DATA_REGISTRY)
+
+    def __getattr__(self, id):
+        """Return one areas states by id."""
+        if self._area_registry.async_get_area(id):
+            return AreaStates(self._hass, id)
+
+    __getitem__ = __getattr__
+
+    def __iter__(self):
+        """Return the iteration over all the areas."""
+        for area in self._area_registry.async_list_areas():
+            yield AreaStates(self._hass, area.id)
+
+    def __call__(self, id):
+        """Return one areas states by id."""
+        return self[id]
+
+    def __repr__(self) -> str:
+        """Representation of areas."""
+        return "<template Areas>"
+
+
+class AreaStates:
+    """Class to expose states of entities belonging to certain area."""
+
+    def __init__(self, hass, id):
+        """Initialize the area state."""
+        self._hass = hass
+        self._id = id
+        self._entity_registry = hass.data.get(entity_registry.DATA_REGISTRY)
+        self._device_registry = hass.data.get(device_registry.DATA_REGISTRY)
+        self._area_registry = hass.data.get(area_registry.DATA_REGISTRY)
```

Comment:
This is a bit hacky.
There's currently no getters for the registers which do not also load them from .storage if that's not already done. That's a behavior we actually **don't** want in this case since `hass` is supposed to be frozen for template evaluation.

Would it be better to make new functions in each registry for this purpose? I foresee few other use cases for such a function.

Reply:
We need to change startup to just load these registries. We can then move to make it a `@callback` function. Maybe an `initialize_template_environment` function ?